yt_dlp/extractor/soundcloud.py

```diff
@@ -656,6 +665,11 @@ def _extract_set(self, playlist, token=None):
                     'playlistId': playlist_id,
                     'playlistSecretToken': token,
                 }, headers=self._HEADERS)
+        album_info = traverse_obj(playlist, {
+            'album': ('title', {str}),
+            'album_artist': ('user', 'username', {str}),
+            'album_type': ('set_type', {lambda x: x or 'playlist'}),
```

Comment:
still wanna type check though
```suggestion
            'album_type': ('set_type', {str}, {lambda x: x or 'playlist'}),
```